src/api/app/services/package_controller_service/url_generator.rb

```diff
@@ -0,0 +1,58 @@
+module PackageControllerService
+  class URLGenerator
+    def initialize(params = {})
+      @project = params[:project]
+      @repository = params[:repository]
+      @arch = params[:arch]
+      @filename = params[:filename]
+      @user = params[:user]
+      @package = params[:package]
+    end
+
+    def logger
+      Rails.logger
+    end
+
+    def rpm_url
+      get_frontend_url_for(controller: 'build') +
+        "/#{@project.name}/#{@repository.name}/#{@arch}/#{@package.name}/#{@filename}"
```

Comment:
[nitpick] `.name` is not needed

Reply:
Hi @Ana06 , I will think about it, I personally find confusing that we implement ```to_s``` in some models but not all, then it's not clear in the method using it, if it receives an object or a string.

Reply:
done